### ISSUE TITLE
Road to 1.0

### ISSUE THREAD

**Issue by timholy** [JuliaGraphics/Colors.jl#361]
I'd like to move the whole JuliaImages stack to version numbers >= 1.0 (see JuliaImages/Images.jl#825). Let's use this issue to collect plans for breaking changes in the near future.

Issues/PRs on my agenda:
- #360 
- #353
- possibly #349, but I worry that will take a while and hence might be 2.0 material
- #273 (ugh)

**Comment by kimikage** [comment]
- A part of #354?

Actually, I have forgotten how far I have worked :sweat_smile:, but somethings should have already been achieved.
https://github.com/JuliaGraphics/Colors.jl/compare/master...kimikage:cnvt

However, there are few breaking changes other than the bug fixes. In regards to the white point `wp`,it follows the current interface, but it is necessary to discuss how white points should be handled.

**Comment by timholy** [comment]
You have done wonders for these packages.

"Cleanups" can be done after 1.0 if they don't break the public API. If we grossly change the result of a white-point aware conversion, though, that is certainly breaking.

**Comment by kimikage** [comment]
- #371 following #353

**Comment by kimikage** [comment]
CC @cormullion
- Tidy the logo up (cf. https://github.com/JuliaLang/julia-logo-graphics)

It may be useful for the testing to add "julia_blue", "julia_green", "julia_purple" and "julia_red" to the named colors. I also feel that "nationalism" is too much, though. :smile:

**Comment by cormullion** [comment]
No problem. What do you mean by "I also feel that "nationalism" is too much, though."?

**Comment by kimikage** [comment]
Adding "julia_blue" etc. to named colors means that it is not necessarily a *public* interest.
"Colors of the Julia users, by the Julia users, for the Julia users!"

**Comment by cormullion** [comment]
Ah, I see. :) Anyway, probably not many people use them (apart from me, I expect)...

**Comment by kimikage** [comment]
Reminder: https://github.com/JuliaGraphics/Colors.jl/issues/353#issuecomment-570967804

**Comment by kimikage** [comment]
- [ ] #496

**Comment by jonathanBieler** [comment]
If there breaking changes in master currently ? I wanted to use some of the XYZ to RGB conversion but last release was before it was added (27 Apr 2021). Would it be possible to make minor release before 1.0 ?

**Comment by timholy** [comment]
Yeah, we need to do something about this. I am pretty swamped with TTFX work until at least mid-October, unfortunately. If you'd be willing to figure out what to do perhaps we could review your work and then make a release.

**Comment by stevengj** [comment]
Can we transition this package to 1.0 already?  The problem with 0.xx releases is that they provide strictly less information than 1.x.y release — semver assumes that every 0.xx release is breaking.'

Version numbers are cheap.  30+ packages depend on this package.  It's been out for ages, and the API hasn't changed much for a while.

**Comment by juliohm** [comment]
As @stevengj said, it would be great to have Colors.jl v1.0.

Who could you make this decision?

**Comment by timholy** [comment]
The only change I suspect I'd want to make is to open the types up for AD (automatic differentiation). I think they may be currently they are too restrictive to permit that, at least for some AD packages.

But I haven't had my eye on this ecosystem for a while now, so however it happens will require that someone do a careful analysis of outstanding PRs and issues.

**Comment by timholy** [comment]
We might also consider moving to a monorepo (together with ColorTypes and ColorVectorSpace). A monorepo is definitely needed for JuliaImages, maybe less important for this package. But that's orthogonal to any code changes.

**Comment by juliohm** [comment]
A monorepo makes sense for Colors.jl given the stability of the API and the small size of related modules 👍🏽

**Comment by stevengj** [comment]
> The only change I suspect I'd want to make is to open the types up for AD (automatic differentiation). I think they may be currently they are too restrictive to permit that, at least for some AD packages.

Unless someone is actively working on this, I'd leave that for Colors v2.0
